packages/core/src/ObjectContainer.js

```diff
@@ -504,9 +515,50 @@ export default class ObjectContainer {
       }
     }
 
+    if (this._isSpread(name) && entry?.sharedInstance.length) {
+      let spreadEntry = Entry.createFromEntry(entry);
+
+      spreadEntry.sharedInstance = entry.sharedInstance.map(sharedInstance =>
+        this.get(sharedInstance)
+      );
+
+      this._entries.set(name, spreadEntry);
+      return spreadEntry;
+    }
+
     return entry;
   }
 
+  /**
+   * Checks whether the name is marked as optional.
+   *
+   * @param {string} name Name of a constant or alias,
+   *        factory function, class or interface constructor, or a fully
+   *        qualified namespace path.
+   * @return {boolean}
+   */
+  _isOptional(name) {
+    return (
+      name?.[1]?.optional || (typeof name === 'string' && name.includes('?'))
```

Comment:
What happens if I write $dependencies with alias '$My?CustomAlias' or '$MyCustomAlias?'?

Reply:
you should not do that :D however I agree that we should check start of the string and probably add some kind of validation for invalid characters? Not sure you want `?` in the middle of an alias.